pkg/operator/quorumguardcleanup/quorumguardcleanupcontroller.go

```diff
@@ -0,0 +1,238 @@
+			Reason: "AsExpected",
+		}))
+	return updateErr
+}
+
+// cleanupOldQuorumGuard checks to see if the new static controller guard pods are ready
+// and then deletes the "etcd-quorum-guard" deployment and PodDisruptionBudget
+func (c *QuorumGuardCleanupController) cleanupOldQuorumGuard(ctx context.Context, syncCtx factory.SyncContext) error {
+	isTargetVersion, err := c.checkClusterOperatorVersion()
+	if err != nil {
+		return fmt.Errorf("failed to ensure clusteroperator version: %w", err)
+	}
+	if !isTargetVersion {
+		syncCtx.Queue().AddAfter(syncCtx.QueueKey(), 1*time.Minute)
+		return nil
+	}
+
+	newGuardLabelSelector, err := labels.Parse(newGuardLabelSelectorString)
+	if err != nil {
+		return fmt.Errorf("failed to parse label selector: %w", err)
+	}
+	newGuardPods, err := c.podLister.Pods(operatorclient.TargetNamespace).List(newGuardLabelSelector)
+	if err != nil {
+		return fmt.Errorf("failed to get etcd static controller guard pods: %w", err)
+	}
+	masterNodeLabelSelector, err := labels.Parse(masterNodeLabelSelectorString)
+	if err != nil {
+		return fmt.Errorf("failed to parse label selector: %w", err)
+	}
+	masterNodes, err := c.masterNodeLister.List(masterNodeLabelSelector)
+	if err != nil {
+		return fmt.Errorf("failed to list master nodes: %w", err)
+	}
+	// TODO: The lister's cache can be empty and return 0 master nodes and trigger a premature cleanup
+	// Find out how to enforce the cache to sync before this controller runs
+	if len(masterNodes) == 0 {
+		return nil
+	}
+
+	// Wait until all new guard pods are rolled out to all master nodes, and have the ready condition
+	numReadyPods := countReadyPods(newGuardPods)
+	if numReadyPods != len(masterNodes) {
+		klog.V(2).Infof("%d/%d guard pods ready. Waiting until all new guard pods are ready", numReadyPods, len(masterNodes))
+		return nil
+	}
+
+	if err := c.removeDeployment(ctx, syncCtx); err != nil {
+		return err
+	}
+	if err := c.removePDB(ctx, syncCtx); err != nil {
+		return err
+	}
+	return nil
+}
+
+// checkClusterOperatorVersion ensures the operator version in clusteroperator/etcd matches the desired version
+// in clusterversion/version
+// i.e all operands have been rolled out to the new image
+func (c *QuorumGuardCleanupController) checkClusterOperatorVersion() (bool, error) {
+	etcdClusterOperator, err := c.clusterOperatorLister.Get("etcd")
+	if err != nil {
+		return false, fmt.Errorf("failed to get clusteroperator/etcd: %w", err)
+	}
+	operatorVersion := ""
+	for _, version := range etcdClusterOperator.Status.Versions {
+		if version.Name == "operator" {
+			operatorVersion = version.Version
+		}
+	}
+	if len(operatorVersion) == 0 {
+		return false, fmt.Errorf("failed to find operator version in clusteroperator/etcd")
+	}
+
+	clusterVersion, err := c.clusterVersionLister.Get("version")
+	if err != nil {
+		return false, fmt.Errorf("failed to check clusterversion/version: %w", err)
+	}
+	if operatorVersion != clusterVersion.Status.Desired.Version {
```

Comment:
you want to compare against this version https://github.com/openshift/cluster-etcd-operator/blob/master/pkg/operator/starter.go#L187

You should pass it through to your controller constructor.